packages/permission-controller/src/PermissionController.ts

```diff
@@ -1573,8 +1581,26 @@ export class PermissionController<
       >
     >;
 
+    const populatedRequest =
+      this.permissionTree.getPopulatedRequest(approvedPermissions);
+
+    const sideEffects = this.getSideEffects(populatedRequest);
```

Comment:
I'm not sure I understand the reasoning behind this. Side-effects were intentionally previously only executed for requested permissions 🤔

Reply:
The side-effect is defined in the permission spec, the child permission is requested implicitly, I'm not sure what's the issue?